src/ui/pages/Notifications/components/EarlierNotification.tsx

```diff
@@ -0,0 +1,102 @@
+import { IonButton, IonList } from "@ionic/react";
+import {
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useMemo,
+  useState,
+} from "react";
+import InfiniteScroll from "react-infinite-scroll-component";
+import { KeriaNotification } from "../../../../core/agent/agent.types";
+import { i18n } from "../../../../i18n";
+import { NotificationItem } from "../NotificationItem";
+import {
+  EarlierNotificationProps,
+  EarlierNotificationRef,
+} from "./EarlierNotification.types";
+
+const SHOWN_EARLIER_NOTIFICATION = 3;
+const LOAD_EARLIER_NOTIFICATION = 5;
+
+const EarlierNotification = forwardRef<
+  EarlierNotificationRef,
+  EarlierNotificationProps
+>(({ data, onNotificationClick, onOpenOptionModal, pageId }, ref) => {
+  const [displayLength, setDisplayLength] = useState(
+    SHOWN_EARLIER_NOTIFICATION
+  );
+
+  const displayNotificationsEarlier = useMemo(() => {
+    if (displayLength >= data.length) return data;
+
+    return data.slice(0, displayLength);
+  }, [displayLength, data]);
+
+  const shouldDisplayExpandNotificationsEarlierButton =
+    data.length > displayLength && displayLength === SHOWN_EARLIER_NOTIFICATION;
+
+  useImperativeHandle(ref, () => ({
+    reset: () => {
+      setDisplayLength(SHOWN_EARLIER_NOTIFICATION);
+    },
+  }));
+
+  const loadMore = useCallback(() => {
+    setDisplayLength((value) => value + LOAD_EARLIER_NOTIFICATION);
```

Comment:
Just to be clear, are you incrementing that by 5? In theory we should just show them all, not increment.

Also, in my test when I was expanding them and navigate to another tab (i.e. Identifiers) and then back to Notifications, they will still be expanded as if this was still in the memory (because the page is still in the DOM), we need them to be collapsed when we exit the page.

Reply:
` In theory we should just show them all, not increment`.

I don't understand this. If we load all after expand I don't think we need use infinite load technical. 

Infinite load technical used to optimize performance of large list. We will load each part of list when user scroll to end.

displayLength used to calculate how many item should be display on screen not total item of earlier notification data.

In here I just load next 5 items when user scroll to end of list until all items of the list has been loaded.